test/multicall/Multicall.js

```diff
@@ -0,0 +1,319 @@
+            const n = 10;
+            const {callData, data} = prepareLongInputData(n, 'processLongInput');
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(2660);    // data is 2.6 kb
+
+            const res = await multicallProcessLongInput(callData);
+            expect(res).to.exist;
+            expect(res.length).to.eq(n);
+            for (let i = 0; i < n; i++) {
+                expect(res[i].success).to.eq(true)
+                expect(res[i].returnData).to.eq(RESULT_FIVE)
+            }
+        });
+
+        it('should be able to make processLongInput calls with length 1000', async function () {
+            const n = 1000;
+            const {callData, data} = prepareLongInputData(n, 'processLongInput');
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(266000);    // data is 260 kb
+
+            const res = await multicallProcessLongInput(callData);
+            expect(res).to.exist;
+            expect(res.length).to.eq(n);
+            for (let i = 0; i < n; i++) {
+                expect(res[i].success).to.eq(true)
+                expect(res[i].returnData).to.eq(RESULT_FIVE)
+            }
+        });
+
+        it('should NOT be able to make processLongInput calls with length 5000 (input size > 1mb)', async function () {
+            const n = 5000;
+            const {callData, data} = prepareLongInputData(n, 'processLongInput');
+
+            const dataSize = getInputLengthInBytes(data);
+            expect(dataSize).to.be.eq(1330000);    // data is 1,3 mb
+            let hasError = false;
+
+            try {
+                const res = await multicallProcessLongInput(callData);
+            } catch (e) {
+                hasError = true;
+            }
+
+            expect(hasError).to.eq(true);
+        });
+
+
+        it('should be able to make processLongInput calls with length 10 and the last call reverts', async function () {
+            let hasError = false;
+            try {
+                const {callData, data} = prepareLongInputData(10, 'processLongInput', true);
+                const res = await multicallProcessLongInput(callData);
+            } catch (e) {
+                hasError = true;
+                expect(e.data).to.exist;
+                expect(e.data).to.eq(REVERT_REASON_DATA);
+            }
+
+            expect(hasError).to.eq(true);
+        });
+    })
+
+    describe('static calls with large output', async function () {
+        it('should be able to make processLongOutput calls with length 13 kb', async function () {
+            const n = 10;
+            const res = await multicallProcessLongOutput(n);
+            expect(res).to.exist;
+            expect(res.length).to.eq(n);
+            const bytes = getOutputLengthInBytes(res.map(r => r.returnData));
+            expect(bytes).to.eq(13440); // 13 kb
```

Comment:
Again, here I don't know where 13440 comes from, same for 824320 in the next test.  At least a comment?